lib/modules/onc_program_procedure.yml

```diff
@@ -0,0 +1,1219 @@
+          * “client_id”; and
+          * Authorization header including “client_id” and “client_secret
+        inferno_supported: yes
+        inferno_tests:
+          - SPB-OSLS-07
+          - EHB-OELS-09
+      - index: 15
+        SUT: |
+          [Both] The health IT developer demonstrates the ability of the
+          Health IT Module to return a JSON object to applications according
+          to the implementation specification adopted in § 170.215(a)(3)
+          and standard adopted in § 170.215(b), including the following:
+          * “access_token”;
+          * “token_type”;
+          * “scope”;
+          * “id_token”;
+          * “refresh_token” (valid for a period of no shorter than three months);
+          * HTTP “Cache-Control” response header field with a value of “no-store”;
+          * HTTP “Pragma” response header field with a value of “nocache”;
+          * “patient” (to support “context-ehr-patient” and “contextstandalone-patient” “SMART on FHIR Core Capabilities”);
+          * “need_patient_banner” (to support “context-banner” “SMART on FHIR Core Capability” for EHR-Launch mode only); and
+          * “smart_style_url” (to support “context-style” “SMART on FHIR Core Capability” for EHR-Launch mode only).
+        TLV: |
+          [Both] The tester verifies the ability of the
+          Health IT Module to return a JSON object to applications according
+          to the implementation specification adopted in § 170.215(a)(3)
+          and standard adopted in § 170.215(b), including the following:
+          * “access_token”;
+          * “token_type”;
+          * “scope”;
+          * “id_token”;
+          * “refresh_token” (valid for a period of no shorter than three months);
+          * HTTP “Cache-Control” response header field with a value of “no-store”;
+          * HTTP “Pragma” response header field with a value of “nocache”;
+          * “patient” (to support “context-ehr-patient” and “contextstandalone-patient” “SMART on FHIR Core Capabilities”);
+          * “need_patient_banner” (to support “context-banner” “SMART on FHIR Core Capability” for EHR-Launch mode only); and
+          * “smart_style_url” (to support “context-style” “SMART on FHIR Core Capability” for EHR-Launch mode only).
+        inferno_supported: yes
+        inferno_tests: 
+          - SPB-OSLS-08
+          - EHB-OELS-10
+      - index: 16
+        SUT: |
+          [Both] The health IT developer demonstrates the ability of the
+          Health IT Module to deny an application’s authorization request in
+          accordance with the implementation specification adopted in §
+          170.215(a)(3).
+        TLV: |
+          [Both] The tester demonstrates the ability of the
+          Health IT Module to deny an application’s authorization request in
+          accordance with the implementation specification adopted in §
+          170.215(a)(3).
+        inferno_supported: partial
+        inferno_notes: |
```

Comment:
I was just confused what you mean by 'ability .. to deny an application's authorization request'.  What is denial here?  We check that they can only choose to accept a subset of the resources for authorization, but I'm not sure that is what is meant here.  Are you referring to the `access_denied` here: https://tools.ietf.org/html/rfc6749#section-4.1.2.1. Now that I look closer, I think that's what is meant, since I'm guessing this step came from here, which discusses error reponses for denial http://www.hl7.org/fhir/smart-app-launch/#step-2-ehr-evaluates-authorization-request-asking-for-end-user-input.

In order to do that, we'll have to set up a 'scenario' that tells the tester to now purposefully fully deny the request, which we don't do yet.